File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.java
import java.util.*;

class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int answer = 0;

        List<Integer> bridge = new ArrayList<>();
        for (int i = 0; i < bridge_length; i++) {
            bridge.add(0);
        }
        
        List<Integer> tw = new ArrayList<>();
        for (int i = 0; i < truck_weights.length; i++) {
            tw.add(truck_weights[i]);
        }
        
        int sb = 0;
        while(tw.size() != 0 || sb != 0) {
            answer += 1;
            if (tw.size() == 0) {
                sb -= bridge.remove(0);
                bridge.add(0);
            } else if (sb - bridge.get(0) + tw.get(0) <= weight) {
                sb -= bridge.remove(0);
                sb += tw.get(0);
                bridge.add(tw.remove(0));
            } else {
                sb -= bridge.remove(0);
                bridge.add(0);
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    bridge = [0] * bridge_length
    
    while len(truck_weights) != 0 or sum(bridge) != 0:
        answer += 1
        if len(truck_weights) == 0:
            bridge.pop(0)
            bridge.append(0)
        elif sum(bridge) - bridge[0] + truck_weights[0] <= weight:
            bridge.pop(0)
            bridge.append(truck_weights.pop(0))
        else:
            bridge.pop(0)
            bridge.append(0)
            
    return answer